### PR TITLE
End-to-end UI tests with Selenium

### PR DESCRIPTION
**Pull request by aliabd**
Tests for four demos (`Image` to `Image` with `live=True`, `Textbox` to `Label`, multiple inputs, `HighlightedText`, `Dropdown`, `Radio`, `Checkbox`, `Slider`, `CheckboxGroup`)

- A lot of the commits are just debugging Selenium with CircleCI
- Will add more demo tests in a future PR

### CODE REVIEW

File: test/test_demos.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        elem = WebDriverWait(driver, timeout).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR,
+                                            ".submit"))
+        )
+        elem.click()
+        elem = WebDriverWait(driver, timeout).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR,
+                                            ".output_interface[interface_id='2'] .output_text"))
+        )
+        while elem.text == "":
```
Comment (aliabid94) [inline test/test_demos.py:102]: can we set a maximum time to this infinite while loops?

File: test/test_demos.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        while not return_dict:
+            time.sleep(0.2)
+
+        URL = LOCAL_HOST.format(return_dict["port"])
+        wait_for_url(URL)
+
+        driver = webdriver.Chrome()
+        driver.set_window_size(1200, 800)
+        driver.get(URL)
+        timeout = 10
```
Comment (aliabid94) [inline test/test_demos.py:131]: seems like everything up till here is common between tests, move to setUp?

File: test/test_demos.py (hunk 1 above)
Comment (aliabd) [reply test/test_demos.py]: Good point, added a time limit on this and all while loops in https://github.com/gradio-app/gradio/pull/72/commits/85c368387660a82c001ddb954b974e5d14826071

File: test/test_demos.py (hunk 2 above)
Comment (aliabd) [reply test/test_demos.py]: Couldn't figure out how to pass a parameter to setUp for the specific demo thread, so opted to refactor with a regular function. See https://github.com/gradio-app/gradio/pull/72/commits/200d7031b5c8713c3a6b1914e67fd9aa22e633a7 and https://github.com/gradio-app/gradio/pull/72/commits/200d7031b5c8713c3a6b1914e67fd9aa22e633a7

**Pull request merged**